### PR TITLE
[volumio] Removing compiler warnings

### LINKED ISSUE #15380: [volumio] removing compiler warning
Removing compiler warnings for better readable

### PR DESCRIPTION
**Pull request by miloit**
Removing compiler warnings

Fixes #15380

### CODE REVIEW

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 1]
```diff
@@ -170,8 +172,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+            logger.debug("Ignoring system command, because device is offline.");
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
+            }
+            return;
+        }
+
         if (command instanceof StringType) {
-            volumio.sendSystemCommand(command.toString());
+            volumiolocal.sendSystemCommand(command.toString());
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:187]:
```suggestion
            VolumioService volumioLocal = volumio;
```

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 2]
```diff
@@ -188,41 +201,73 @@ private void clearChannels() {
     }
 
     private void handleVolumeCommand(Command command) {
+        @Nullable
+        VolumioService volumiolocal = volumio;
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:205]:
```suggestion
        VolumioService volumioLocal = volumio;
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Some small comments, half way i stopped as the feedback would be the same:
- check instanceof 
- remove logger when allready setting ThingStatusDetail
- use pascal casing vars
- inline null annotation should not be needed

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 3]
```diff
@@ -286,17 +353,28 @@ public void initialize() {
 
     @Override
     public void dispose() {
-        if (volumio != null) {
+        @Nullable
+        VolumioService volumiolocal = volumio;
+
+        if (volumiolocal == null) {
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:359]: Remove this block

Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:359]: removed

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 4]
```diff
@@ -188,41 +201,73 @@ private void clearChannels() {
...
+            logger.debug("Ignoring volume command, because device is offline.");
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
+            }
+            return;
+        }
+
         if (command instanceof PercentType) {
```
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:215]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 5]
```diff
@@ -188,41 +201,73 @@ private void clearChannels() {
     }
 
     private void handleVolumeCommand(Command command) {
+        @Nullable
+        VolumioService volumiolocal = volumio;
+
+        if (volumiolocal == null) {
+            logger.debug("Ignoring volume command, because device is offline.");
```
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:208]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 6]
```diff
@@ -66,9 +66,11 @@ public VolumioHandler(Thing thing) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        @Nullable
+        VolumioService volumiolocal = volumio;
```
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:70]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 7]
```diff
@@ -66,9 +66,11 @@ public VolumioHandler(Thing thing) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        @Nullable
```
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:69]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 8]
```diff
@@ -170,8 +172,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     }
 
     private void sendSystemCommand(Command command) {
+        @Nullable
+        VolumioService volumiolocal = volumio;
```
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:176]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 9]
```diff
@@ -170,8 +172,19 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+
+        if (volumiolocal == null) {
+            logger.debug("Ignoring system command, because device is offline.");
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
```
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:181]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 1 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:187]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 2 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:205]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 10]
```diff
@@ -66,10 +66,11 @@ public VolumioHandler(Thing thing) {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        VolumioService volumioLocal = volumio;
         logger.debug("channelUID: {}", channelUID);
 
-        if (volumio == null) {
-            logger.debug("Ignoring command {} = {} because device is offline.", channelUID.getId(), command);
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized.");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:73]: Please remove the logger and improve the `ThingStatusDetail`

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 11]
```diff
@@ -188,41 +199,70 @@ private void clearChannels() {
     }
 
     private void handleVolumeCommand(Command command) {
-        if (command instanceof PercentType) {
-            volumio.setVolume((PercentType) command);
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:205]: See previous comment

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 12]
```diff
@@ -188,41 +199,70 @@ private void clearChannels() {
...
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:207]: See previous comment

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 13]
```diff
@@ -188,41 +199,70 @@ private void clearChannels() {
...
         } else {
             logger.error("Command is not handled");
         }
     }
 
     private void handleStopCommand(Command command) {
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:225]: See previous comment

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 14]
```diff
@@ -188,41 +199,70 @@ private void clearChannels() {
...
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:227]: See previous comment

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 15]
```diff
@@ -188,41 +199,70 @@ private void clearChannels() {
...
         } else if (command.equals(RefreshType.REFRESH)) {
             updateState(VolumioBindingConstants.CHANNEL_STOP, UnDefType.UNDEF);
         }
     }
 
     private void handlePlaybackCommands(Command command) {
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:244]: Please remove logger and improve `ThingStatusDetail`

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 16]
```diff
@@ -246,9 +286,19 @@ private void handlePlaybackCommands(Command command) {
      * - PUSH.STATE -
      */
     private void bindDefaultListener() {
-        volumio.on(Socket.EVENT_CONNECT, connectListener());
-        volumio.on(Socket.EVENT_DISCONNECT, disconnectListener());
-        volumio.on(VolumioEvents.PUSH_STATE, pushStateListener());
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:292]: Please remove logger and improve `ThingStatusDetail`

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 17]
```diff
@@ -257,6 +307,16 @@ private void bindDefaultListener() {
      */
     @Override
     public void initialize() {
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:312]: Why did you add this block? 
If i'm not mistaken, volumioLocal will always be null, as this method is only called when the class initializes and therefore will allways be null.

**Review by lsiepel — CHANGES_REQUESTED** [review]
Found more occurences of the same issue.

Did you test your code?

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (lsiepel) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:179]:
````suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Handler is in an unknown state");
````

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 10 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:73]: see [other comment](https://github.com/openhab/openhab-addons/pull/15379#discussion_r1288532102)

**Comment by miloit** [comment]
> Found more occurences of the same issue.
> 
> Did you test your code?

Yes, tested and build was with no error

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:179]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 10 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:73]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 12 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:207]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 13 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:225]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 15 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:244]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 16 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:292]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 17 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:312]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 14 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 11 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 18]
```diff
@@ -170,8 +170,18 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     }
 
     private void sendSystemCommand(Command command) {
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Volumio service was not yet initialized, cannot handle send system command.");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:178]:
```suggestion
                        "Volumio service was not yet initialized, cannot send system command.");
```

**Review by lsiepel — COMMENTED** [review]
Better now :-)

**Review by lsiepel — APPROVED** [review]

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 19]
```diff
@@ -257,6 +306,8 @@ private void bindDefaultListener() {
      */
     @Override
     public void initialize() {
+        VolumioService volumioLocal = volumio;
```
Comment (wborn) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: I think you forgot a null check here because the compiler prints:

`Warning:  /home/runner/work/openhab-addons/openhab-addons/bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:[331,17] Potential null pointer access: The variable volumioLocal may be null at this location`

Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: If i'm not mistaken, volumioLocal will always be null, as this method is only called when the class initializes and therefore will allways be null.

See also comment from @lsiepel

Comment (lsiepel) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: That was because a preliminary return caused it never to instantiate. It still needs a check.

Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: Than I need help how can i remove this last compiler warning?? I tried a lot but always get this error...Can you help me?

Comment (wborn) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: Assign it to the local variable and then the local variable to the field

```java 
Volumio service volumioLocal = new VolumioService(protocol, hostname, port, timeout);
volumio = volumioLocal;
```

Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: Getting this error

R] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,58] protocol cannot be resolved to a variable
[ERROR] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,68] hostname cannot be resolved to a variable
[ERROR] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,78] port cannot be resolved to a variable
[ERROR] D:\Projects\openhab\git\openhab-addons\bundles\org.openhab.binding.volumio\src\main\java\org\openhab\binding\volumio\internal\VolumioHandler.java:[309,84] timeout cannot be resolved to a variable

Comment (wborn) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: The idea is to introduce the local var here:

https://github.com/openhab/openhab-addons/blob/11e954847f38d7987899e7d4faad82f1cde674f0/bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java#L327

Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 20]
```diff
@@ -274,10 +324,18 @@ public void initialize() {
...
             try {
                 volumio = new VolumioService(protocol, hostname, port, timeout);
+                VolumioService volumioLocal = volumio;
+                if (volumioLocal == null) {
+                    if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Error during volumio service initialization.");
+                    }
+                    return;
+                }
```
Comment (wborn) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:334]: It can never be null if you write it like this:

```suggestion
                VolumioService volumioLocal = new VolumioService(protocol, hostname, port, timeout);
                volumio = volumioLocal;
```

Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java]: solved

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java [hunk 21]
```diff
@@ -257,6 +306,7 @@ private void bindDefaultListener() {
      */
     @Override
     public void initialize() {
+
```
Comment (lsiepel) [inline bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java:309]: Very minor, but please remove empty line.
You could always check the SAT for this kind of issues.

**Review by lsiepel — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java (hunk 21 above)
Comment (miloit) [reply bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java]: solved

**Review by lsiepel — APPROVED** [review]
LGTM
